std/typecons.d

```diff
@@ -3109,6 +3109,71 @@ struct Nullable(T)
     {
         return isNull ? fallback : _value.payload;
     }
+
+    // Range interface
+    /**
+     * Returns whether or not any values remain in this range.
+     */
+    alias empty = isNull;
+    /**
+     * Advances this range.
+     */
+    alias popFront = nullify;
+    /// ditto
+    alias popBack = nullify;
+    /**
+     * Returns the range's current value.
+     */
+    @property ref inout(T) front() inout @safe pure nothrow
+    {
+        return get();
+    }
+    /// ditto
+    alias back = front;
+    ///
+    @property inout(typeof(this)) save()() inout
+    {
+        return this;
+    }
+    ///
+    inout(typeof(this)) opSlice()() inout
+    {
+        return this;
+    }
+    ///
+    inout(typeof(this)) opSlice()(size_t from, size_t to) inout
```

Comment:
I get the range interface, but not sure why it needs slicing.

Reply:
See previous comment re: `opSlice`.

Reply:
I don't know why. I don't know why not, either. Either way, it was simple enough to support.

Reply:
Prefer less code, it's easier to maintain.

Reply:
Satisfying `hasSlicing!T` allows more efficient branches to be taken in several Phobos algorithms:

* [`std.algorithm.mutation.copy`][1]
* [`std.algorithm.searching.find`][2]
* [`std.range.take`][3]
* etc.

It's also necessary if we want ranges of `Nullables` (e.g., `chain(nullable(x), nullable(y), nullable(z))`) to support slicing.

[1]: https://github.com/dlang/phobos/blob/v2.099.0/std/algorithm/mutation.d#L411
[2]: https://github.com/dlang/phobos/blob/v2.099.0/std/algorithm/searching.d#L1982
[3]: https://github.com/dlang/phobos/blob/v2.099.0/std/range/package.d#L2253